### ISSUE TITLE
Docker images bigger than 5GB are not supported

### ISSUE THREAD

**Issue by PidgeyBE** [suse-edge/edge-image-builder#617]
Hello

I'm trying to package a relatively large docker image with EIB.
The packaged `.tar.zst` file created by Hauler is 5.3 GB.

If I understand correctly, when SLE Micro starts, after it was installed, Hauler loads all airgapped images. This happens by extracting (?) every tar file to `/tmp`. 
It seems that `/tmp` is limited to `4.8G`, making Hauler fail:
```
Nov 19 17:55:06 localhost.localdomain sh[2769]: Error: reading file in tar archive: /tmp/hauler2726428976/blobs/sha256/f0f5fb4389cd3384679b2afb491c6faf24de9562e9991311b5a2b73868de96be: writing file: write /tmp/..491c6faf24de9562e9991311b5a2b73868de96be: no space left on device
Nov 19 17:55:06 localhost.localdomain sh[2769]: Usage:
Nov 19 17:55:06 localhost.localdomain sh[2769]:   hauler store load [flags]
Nov 19 17:55:06 localhost.localdomain sh[2769]: Flags:
Nov 19 17:55:06 localhost.localdomain sh[2769]:   -h, --help             help for load
Nov 19 17:55:06 localhost.localdomain sh[2769]:   -t, --tempdir string   overrides the default directory for temporary files, as returned by your OS.
Nov 19 17:55:06 localhost.localdomain sh[2769]: Global Flags:
Nov 19 17:55:06 localhost.localdomain sh[2769]:       --cache string       (deprecated flag and currently not used)
Nov 19 17:55:06 localhost.localdomain sh[2769]:   -l, --log-level string    (default "info")
Nov 19 17:55:06 localhost.localdomain sh[2769]:   -s, --store string       Location to create store at (default "store")
Nov 19 17:55:06 localhost.localdomain sh[2769]: 2024-11-19 17:55:06 ERR reading file in tar archive: /tmp/hauler2726428976/blobs/sha256/f0f5fb4389cd3384679b2afb491c6faf24de9562e9991311b5a2b73868de96be: writing ... no space left on device
Nov 19 17:55:06 localhost.localdomain systemd[1]: eib-embedded-registry.service: Control process exited, code=exited, status=1/FAILURE
Nov 19 17:55:06 localhost.localdomain systemd[1]: eib-embedded-registry.service: Failed with result 'exit-code'.
Nov 19 17:55:06 localhost.localdomain systemd[1]: Failed to start Load and Serve Embedded Registry.
```

I'm testing in a [VM](https://github.com/suse-edge/edge-image-builder/blob/main/docs/testing-guide.md) and results are the same for 100GB and 250GB disks.

Hauler itself mentions this in their docs: https://rancherfederal.github.io/hauler-docs/docs/known-limits
> When using hauler, it will default to write temporary files to /tmp. If you do not have about 3x the amount of space available in your /tmp compared to the size of the store or the haul, then hauler may fail.

Is there any way I can make the `/tmp` partition larger?
Or can I make Hauler use another temp directory? They do have a `--tempdir` flag...

**Comment by dbw7** [comment]
Sorry for the delay in response, when we combed through the issues I believed this was a different issue than what it is. For `/tmp`, this is actually memory bound as opposed to storage bound.

I will update the documentation to be more descriptive, but what is actually happening is:
I believe the size of `/tmp` is ½ of your system’s memory. So if you have 8GB RAM and your uncompressed container image is 5GB, when Hauler tries to load it into `/tmp` it will fail with no space left on device. 

Unfortunately the only way to fix this is to increase the RAM of your system as Hauler cannot load a unit smaller than a single container image, so it fully depends on the size of the image.

**Comment by PidgeyBE** [comment]
> Unfortunately the only way to fix this is to increase the RAM of your system as Hauler cannot load a unit smaller than a single container image, so it fully depends on the size of the image.

I do not agree: The issue can also be fixed by exposing Hauler's `--tempdir` flag. 
I'm currently working around the issue by disabling the `eib-embedded-registry.service` daemon and replacing it with my own flavour of it, where I use the `--tempdir` flag to avoid `/tmp`.

**Comment by dbw7** [comment]
Oh

**Comment by dbw7** [comment]
You're absolutely right! Originally we were using Hauler v1.0.0 and this wasn’t an option so we had to implement our own workarounds for minimizing the RAM usage. When we upgraded to v1.0.7, I failed to notice this change in the release notes, this is a great find, thank you! I have fixed this in #676 . When you get a chance, let me know if this change works well for you!
